src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONBuilder.java

```diff
@@ -297,11 +301,17 @@ public static int getGeometryType(Geometry geometry) {
     public GeoJSONBuilder value(Object value) {
         if (value instanceof java.util.Date || value instanceof Calendar) {
             value = Converters.convert(value, String.class);
+        } else if (value instanceof Double) {
+            value = roundToNumberOfDecimalPlaces((Double)value);
         }
         super.value(value);
         return this;
     }
-    
+
+    private double roundToNumberOfDecimalPlaces(Double value) {
+        return new BigDecimal((Double)value).setScale(numDecimals, BigDecimal.ROUND_HALF_UP).doubleValue();		
```

Comment:
Looking at this one again, this is a massive generator of garbage, by allocating a large object as BigDecimal to format each and every ordinate. While I haven't tried, I believe this is going to affect performance significantly.
The GML2 output format got a nice speedup when we optimized its decimal places handling, see here: 
https://github.com/geotools/geotools/blob/master/modules/library/main/src/main/java/org/geotools/gml/producer/CoordinateWriter.java#L295